h2o-algos/src/main/java/hex/util/EffectiveParametersUtils.java

```diff
@@ -0,0 +1,68 @@
+package hex.util;
+
+import hex.Model;
+import hex.ScoreKeeper;
+import hex.genmodel.utils.DistributionFamily;
+import hex.tree.SharedTreeModel;
+
+
+public class EffectiveParametersUtils {
```

Comment:
I don't think it should be checked here. It works like if (evaluate_auto) {initActualParamValues()} and in initActualParamValues you set values either by some method from EffectiveParametersUtils.java or by actual commands if it is something model specific.

params._evaluate_auto is a safety net. Switcher to be able to use the before-this-change version of param evaluation easily, in case we find out something is being set incorrectly and it messes up calculation. Maybe it really could be done on some different level. Do you know whether we have some similar switcher so that it could be done in a similar pattern?

Reply:
In that case, I think it makes more sense to enable/disable this globally as a system property.
See `SQLManager.shouldUseTempTable()` or `automl.Algo.enabled()` for examples.
Just be careful to never store the value of this system property in a class field or in a singleton: user should be able to change the value of this property dynamically without having to restart the backend, that's why I was also suggesting the client `backend property` API to switch those easily.

Reply:
f76ba39